pandas/core/dtypes/concat.py

```diff
@@ -45,6 +46,8 @@ def get_dtype_kinds(l):
             # if to_concat contains different tz,
             # the result must be object dtype
             typ = str(arr.dtype)
+        elif is_range(arr):
```

Comment:
I'm fine with using ``isinstance(arr, ABCRangeIndex)`` in ``get_dtype_kinds``... but I'm not special casing ``Index._concat``: I'm only overwriting ``RangeIndex._append_same_dtype``. So I need ``get_dtype_kinds`` to (also) distinguish ``RangeIndex``. Isn't the point of ``get_dtype_kinds`` precisely to distinguish stuff which can be concatenated "natively" together?!